src/core/ddsc/src/dds_write.c

```diff
@@ -71,25 +72,26 @@ static void *create_iox_chunk(dds_writer *wr)
     iceoryx_header_t *ice_hdr;
     void *sample;
     uint32_t sample_size = wr->m_topic->m_stype->iox_size;
-    uint32_t chunk_size = DETERMINE_ICEORYX_CHUNK_SIZE(sample_size);
     while (1)
     {
-      enum iox_AllocationResult alloc_result = iox_pub_loan_chunk(wr->m_iox_pub, (void **) &ice_hdr, chunk_size);
+      enum iox_AllocationResult alloc_result = 
+        iox_pub_loan_aligned_chunk_with_user_header(wr->m_iox_pub, &sample, sample_size, IOX_C_CHUNK_DEFAULT_USER_PAYLOAD_ALIGNMENT, sizeof(iceoryx_header_t), 8);
+      iox_chunk_header_t * iox_chunk_header = iox_chunk_header_from_user_payload(sample);
+      ice_hdr = (iceoryx_header_t*)iox_chunk_header_to_user_header(iox_chunk_header);
```

Comment:
I think there is no function to get the user header directly from the sample, is there?

Could be changed in iceoryx API in the next release. But if we do so we should also think about whether we really want to expose the internal chunk header in the user API (I would restrict it to a private header).